crates/db/src/db_worldstate.rs

```diff
@@ -112,6 +113,44 @@ impl DbWorldState {
         }
         Ok(())
     }
+
+    fn check_chparent_property_conflict(
+        &self,
+        perms: &Obj,
+        obj: &Obj,
+        new_parent: &Obj,
+    ) -> Result<(), WorldStateError> {
+        // If object or one of its descendants defines a property with the same name as one defined
+        // either on new-parent or on one of its ancestors, then E_INVARG is raised.
+        let obj_or_descendant_props = self
+            .descendants_of(perms, obj, true)?
+            .iter()
+            .map(|descendant| self.get_tx().get_properties(&descendant))
+            .collect::<Result<Vec<_>, _>>()?
+            .into_iter()
+            .flatten();
+        let new_parent_or_ancestors_props = self
+            .ancestors_of(perms, new_parent, true)?
+            .iter()
+            .map(|ancestor| self.get_tx().get_properties(&ancestor))
+            .collect::<Result<Vec<_>, _>>()?
+            .into_iter()
+            .flatten()
+            .collect::<Defs<_>>();
```

Comment:
We'll iterate over this multiple times, so have to `collect` it. Might as well use `Defs` as the container. Unsure if `Vec` would be (a) more "correct" (b) more performant.

Reply:
... Possible micro-optimization: if we generally expect that the new parent will have more ancestors-properties, and the object being moved will have less descendant-properties, then we can flip these around, and collect the descendant-props instead of the ansector-props.